### PR TITLE
Add NPC mutation rounds.

### PR DESCRIPTION
**Pull request by SpencerMichaels**
This PR is split off from #22487.

It adds a new tag, `mutation_rounds`, to NPC class templates. This allows you to cause the generated NPC to have mutated along one or more mutation categories an arbitrary number of times. This means that if, say, you want an Insect mutant NPC, you can make use of the existing mutation system to make him undergo several rounds of `MUTCAT_INSECT` mutations before spawn.

You can specify mutation rounds in two ways; both use the RNG distribution objects like, for instance, the `bonus_str` field. The first is a shortcut for general mutations (like what you'd get from drinking regular mutagen).
```
    {
        "type" : "npc_class",
        ... more fields here ...
        "mutation_rounds" : { "rng" : [ 1, 3 ] }
    },

```
The second lets you specify any number of categories. In this example, our NPC always undergoes 1 `ANY` mutation, as well as 2d3 `INSECT` mutations before spawn.
```
    {
        "type" : "npc_class",
        ... more fields here ...
        "mutation_rounds" : [
            [ "MUTCAT_ANY", { "constant" : 1 } ],
            [ "MUTCAT_INSECT", { "rng" : [ 2, 3 ] } ]
        ]
    },
```
Taken together with the trait group system (see #22487, separate PR pending) you can use the groups to generate NPCs with certain fixed traits and then mutate them to push them further along that semirandom mutation path.

# Screenshots
This is an NPC mutated at spawn via the `MUTCAT_INSECT` category example above. (He also spawned with some starting traits, which I added via my trait groups system.)

This is an NPC mutated at spawn with 10 rounds of `MUTCAT_RAPTOR`.

### CODE REVIEW

File: src/npc_class.cpp [hunk 1]
```diff
@@ -244,6 +244,26 @@ void npc_class::load( JsonObject &jo, const std::string & )
         }
     }
 
+    /* Mutation rounds can be specified as either a single distribution:
+     *   "mutation_rounds" : { "constant" : 1 }
+     * or a map of categories to distributions:
+     *   "mutation_rounds" : [
```
Comment (kevingranade) [inline src/npc_class.cpp]: This example doesn't match the code, it looks more like:
```
"mutagen_rounds": [
  [ "MUTCAT_ANY", { "constant": 1 } ],
  [ "MUTCAT_INSECT", { "rng": [1, 3] } 
]
```
I prefer the syntax in the comment, is there a reason you changed it?

**Comment by kevingranade** [comment]
Other than the one comment about json syntax, this looks good.  I'll wait to merge until you get back to me with that detail though.

**Comment by sxmichaels** [comment]
Yep, I just saw that — just fixed it. (I noticed also that the CI checks failed for my last two commits, but it looks like they were package manager errors on the CI host, not errors in the build itself...)

**Comment by sxmichaels** [comment]
Oh, if that's the case I can put up another PR to make them objects — I didn't realize you were suggesting changing the structure earlier. Should take just a few minutes.

**Pull request merged**